core/state_transition.go

```diff
@@ -741,6 +759,13 @@ func (st *stateTransition) refundGas(refundQuotient uint64) uint64 {
 		st.evm.Config.Tracer.OnGasChange(st.gasRemaining, 0, tracing.GasChangeTxLeftOverReturned)
 	}
 
+	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && !st.msg.IsDepositTx {
+		// Return ETH to transaction sender for operator cost overcharge.
+		if actualCost := st.evm.Context.OperatorCostFunc(new(big.Int).SetUint64(st.gasUsed()), st.evm.Context.Time); actualCost != nil {
+			st.state.AddBalance(st.msg.From, new(uint256.Int).Sub(worstCaseCost, actualCost), tracing.BalanceIncreaseGasReturn)
```

Comment:
This doesn't quite make sense. Above on L669, we reward the entire worst case operator fee to the operator fee recipient. But then, we refund `worstCase - actual`? It seems like this would mint new ETH out of thin air.

Reply:
Maybe the variable are badly named, they are originated from [the comment above](https://github.com/ethereum-optimism/op-geth/pull/388#discussion_r1941598508)

worstCase is gasLimit, and actual is gasUsed, so we refound the remaining gas

Reply:
@clabby is referring to the bug originally found by @tynes that the amount being credited to the operator does not match the operator component of the amount being debited from the user.

Reply:
> worstCase is gasLimit, and actual is gasUsed, so we refund the remaining gas

yep, it looks like the gas refund is correct. But when we credit the operator fee recipient on L669, we don't give them `actual`, we give them `worstCase`. This would create a situation where new ETH is created on the L2 (which can not happen, except in deposit transactions).

What we want here is to credit:
* The fee recipient `actual`
* The sender `worstCase - actual` (remaining)